modules/transactionpool/transactionpool.go

```diff
@@ -60,13 +60,17 @@ type (
 		transactionSets     map[TransactionSetID][]types.Transaction
 		transactionSetDiffs map[TransactionSetID]modules.ConsensusChange
 		transactionListSize int
-		blockHeight         types.BlockHeight
 		// TODO: Write a consistency check comparing transactionSets,
 		// transactionSetDiffs.
 		//
 		// TODO: Write a consistency check making sure that all unconfirmedIDs
 		// point to the right place, and that all UnconfirmedIDs are accounted for.
 
+		blockHeight         types.BlockHeight
+		recentConfirmedFees []feeSummary
+		txnsPerBlock        []uint64       // the number of txns in each of the blocks
```

Comment:
does this grow unbounded? i.e. will it end up with 100k elements? Or does it operate on a much smaller window?